### ISSUE TITLE
[ERROR] HTTPError 403: FORBIDDEN

### ISSUE THREAD

**Issue by Adeleine** [anidl/funimation-downloader-nx#44]
funi -s 147977 -e 5 -q 0 --sub --mks

[INFO] Selected Episodes: 0005

[INFO] The Melancholy of Haruhi Suzumiya - S1E05 - The Melancholy of Haruhi Suzumiya V
[INFO] Available streams (Non-Encrypted):
[#148058] Japanese [Uncut] (selected)
[#148026] English [Uncut]
[ERROR] HTTPError 403: FORBIDDEN

Episode 1-4 downloaded fine. Episode 5, and everything after does not download, and returns a 403 error.

**Comment by louanbastos** [comment]
same error for me.

**Comment by sharinganthief** [comment]
experiencing with 4.7.0-beta.2 as well

**Comment by LonelyShujin** [comment]
Did you verify the video or the version wasn't premium only. If so you'll need to login into a account with premium most likely.

**Comment by LordEliteDragon** [comment]
do u have a funimation subscription or u a free user?

**Comment by fuxxociety** [comment]
> 
> 
> do u have a funimation subscription or u a free user?

Same issue for me - using funimation premium account. Even when simply logging in with 'node funi --auth':
`[ERROR] HTTPError 403: Forbidden`

**Comment by levaculik2** [comment]
someone managed to solve this problem ???

[ERROR] HTTPError 403: Forbidden

**Comment by izu-co** [comment]
@levaculik2 There is nothing to solve.
The request has been rejected by funimation, either because you are not logged in, or because funimation isn't supported in your region.

**Comment by Oujiii** [comment]
This error is 99% because you are trying to download outside a supported region. Just use a proxy.

**Comment by jcdark** [comment]
My region is Brazil, here it has access to Funimation, but even so the Error happens to me too

I believe that it really is a mistake in some situations, it would be good to be checked

**Comment by izu-co** [comment]
> My region is Brazil, here it has access to Funimation, but even so the Error happens to me too
> 
> I believe that it really is a mistake in some situations, it would be good to be checked

Like I stated earlier there is nothing to check. Our request has been blocked by funimation. This usually happens if you are trying to access premium content without being logged in or if funimation is not supported in your region.

We (as the developers) have no control over funimations response to the made request.

**Comment by jcdark** [comment]
> > My region is Brazil, here it has access to Funimation, but even so the Error happens to me too
> > I believe that it really is a mistake in some situations, it would be good to be checked
> 
> Like I stated earlier there is nothing to check. Our request has been blocked by funimation. This usually happens if you are trying to access premium content without being logged in or if funimation is not supported in your region.
> 
> We (as the developers) have no control over funimations response to the made request.

I have premium access, yet I'm getting an error when trying to authenticate

**Comment by jcdark** [comment]
> @jcdark The server used for scrapping the files requires an US IP address or DNS request, I'm also from Brazil I can confirm. You won't be able to login without a VPN, proxy or a Smart DNS. This is outside the scope of this project.

This is useful information @usuario, I hadn’t read this feature anywhere. I will try.

**Comment by izu-co** [comment]
The project is not only US based. I (since I live in Germany) can use it from GB (through a VPN). 
I rewrote many parts of the code and have never encountered a US only part. 

[This](https://prod-api-funimationnow.dadcdigital.com/api/source/funimation/search/auto/?unique=true&limit=100&q=Test&offset=0) is a URL Request generated by the tool. Can you try to open it? (An Image of the result would be helpful :))

**Comment by Oujiii** [comment]
![image](https://user-images.githubusercontent.com/34745265/119507480-413f6980-bd45-11eb-8fdb-496e86737395.png)

I disabled my proxy and this is what it shows. Funimation has been available in the UK for years now, but on Brazil and Mexico only arrived the by end of last year, so this request is probably using an URL not supported on those new regions.

**Comment by izu-co** [comment]
Well every singe request is based on ```https://prod-api-funimationnow.dadcdigital.com/api```. 

I don't know why they won't support Brazil and Mexico under this URL, but as long as they won't the toll cannot be used from those regions.

**Comment by Oujiii** [comment]
Yes, this is what I mentioned. It won't be available without a proxy or anything of the sorts. This API is probably not ready for the new regions, as Funimation does half ass a lot of things. This is unfortunate, but it is how it is.

Did you have a moment to take a look into the subtitle issue I subimted earlier this week?

**Comment by izu-co** [comment]
The one you closed and I resolved?

**Comment by Oujiii** [comment]
No,[ this one](https://github.com/izu-co/funimation-downloader-nx/issues/1) on your repo.

**Comment by jcdark** [comment]
A funimation does things in half, always...

Currently I use VLC to download the videos, but the audio is wrong, as if it lost synchrony in time

**Comment by Oujiii** [comment]
Yes, unfortunately. I think you can close this issue then @jcdark since it's something outside the control of the project maintainers.

**Comment by jcdark** [comment]
> Yes, unfortunately. I think you can close this issue then @jcdark since it's something outside the control of the project maintainers.

Analyzing the site here in Brazil, funimation uses the same api (https://prod-api-funimationnow.dadcdigital.com/api) and the same endpoints for authentication and search, I believe it is some parameter that is missing from the application to be able to authenticate with the countries of Brazil and Mexico

As for closing the issue, I was not the one who reported it ...

**Comment by izu-co** [comment]
> Analyzing the site here in Brazil, funimation uses the same api (https://prod-api-funimationnow.dadcdigital.com/api) and the same endpoints for authentication and search, I believe it is some parameter that is missing from the application to be able to authenticate with the countries of Brazil and Mexico

I will take a look into it, since I am currently working on the downloader.

**Comment by izu-co** [comment]
How did you analyse the site? I would like to run some tests since I fixed the subtitle thing.

**Comment by jcdark** [comment]
> How did you analyse the site? I would like to run some tests since I fixed the subtitle thing.

I analyze the XHR returns. It is in this way that today I download the VTT and M3u8 files to download the videos and subtitles, however it is manually. I wanted to Optimize like I did with CrunchRoll

Regarding the forbidem error, it is linked that the funimotion API application detects that it is a request from a Robot. This happened to me when I tried to make the request (Login) via PHP, but I couldn't solve it there.

----------------------------Login request response-------------------------
<html style="height:100%"><head><META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW"><meta name="format-detection" content="telephone=no"><meta name="viewport" content="initial-scale=1.0"><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"></head><body style="margin:0px;height:100%"><iframe id="main-iframe" src="/_Incapsula_Resource?CWUDNSAI=22&xinfo=6-40026542-0%200NNN%20RT%281621955576061%2081%29%20q%280%20-1%20-1%20-1%29%20r%280%20-1%29%20B15%2811%2c1343935%2c0%29%20U6&incident_id=1239000390225927996-219955029180090822&edet=15&cinfo=0b000000&rpinfo=0" frameborder=0 width="100%" height="100%" marginheight="0px" marginwidth="0px">Request unsuccessful. Incapsula incident ID: 1239000390225927996-219955029180090822</iframe></body></html>

**Comment by Oujiii** [comment]
@jcdark Sorry to ask you here, but do you have a working tool for Crunchyroll? Can you publish the code? None of the automated tools I use are working anymore.

**Comment by izu-co** [comment]
Doesn't @anidl have one?

**Comment by Oujiii** [comment]
Not working for me as in [#46](https://github.com/anidl/crunchyroll-downloader-nx/issues/46)

**Comment by izu-co** [comment]
Well I don't use crunchy roll, so I can't help you.

@jcdark I don't think that it is robot detection, since request from GB or US are completely fine.

```
<div style="display: none;">
        <div><div class="error-title">Access Denied</div>
<div class="error-code">Error 15</div>
<div class="hostname">prod-api-funimationnow.dadcdigital.com</div>
<div class="date">2021-05-25 15:26:07 UTC</div>
</div>
        <div><div class="title">What happened?</div>
<div class="description">This request was blocked by the security rules</div>
<div class="info-container">
	<div class="info">
		<span class="label">Your IP:</span>
		<span class="value">192.154.196.14</span>
	</div>
	<div class="info">
		<span class="label">Proxy IP:</span>
		<span class="value">45.60.73.224</span>
		<span class="parentheses-text"> (ID 10210-100)</span>
	</div>
	<div class="info">
		<span class="label">Incident ID:</span>
		<span class="value">985000410028306871-70961625822726732</span>
	</div>
</div>
```

I think that the new region are just blocked by the security rules or (as you said) some kind of information is missing, however I have no idea which.

**Comment by jcdark** [comment]
![Screenshot from 2021-05-25 12-26-20](https://user-images.githubusercontent.com/1435088/119525924-4d332780-bd55-11eb-9dea-482c4a884de2.png)
![Screenshot from 2021-05-25 12-26-52](https://user-images.githubusercontent.com/1435088/119525930-4e645480-bd55-11eb-831a-7f9a86dbf1e0.png)

**Comment by jcdark** [comment]
It's strange, but I already realized that Funimation has two servers, which work separately.

Some videos that have Brazilian dubbing are not in the American version, although they are dubbed in English

A little confusing this architecture ...
I'll keep trying

**Comment by izu-co** [comment]
Year, I was grateful that the server infrastructure was programmed by anidl.
Thanks for trying tho.

**Comment by izu-co** [comment]
I would assume that this issue has been resolved. If not, reopen it or create a new one, please.

**Comment by sl1pkn07** [comment]
same problem here throught VPN (nord using wireguard protocol (nordlynx) to connect to US (`You are connected to United States #9256`)

using this series #1071625 (Appare-Ranman!) as example, only can download uncut version of episode 2 and 13, the rest get 403, but i can download them if parse the option `--simul`

~~~
[INFO] APPARE-RANMAN! - S1E1 - Sunny [Appare], With Occasional Showers [Kosame]
[INFO] Available streams (Non-Encrypted):
[#1443614] English [Uncut]
[#2948483] Japanese [Uncut] (selected)
[#1164753] English [Simulcast]
[#1071760] Japanese [Simulcast]
[ERROR] HTTPError 403: FORBIDDEN
~~~

~~~
[INFO] APPARE-RANMAN! - S1E1 - Sunny [Appare], With Occasional Showers [Kosame]
[INFO] Available streams (Non-Encrypted):
[#1443614] English [Uncut]
[#2948483] Japanese [Uncut]
[#1164753] English [Simulcast]
[#1071760] Japanese [Simulcast] (selected)
[INFO] Servers available:
~~~

using a free account

maybe is a premium feature for that episodes?

greetings

**Comment by izu-co** [comment]
A Forbidden on funimation can happen if you try to download premium content with a free account.

Also please note that this issue is closed and staled. Please refrain from commenting in these old issues.
